### ISSUE TITLE
Error handler obscures original stack

### ISSUE THREAD

**Issue by dblock** [collectiveidea/delayed_job#673]
The implementation in https://github.com/collectiveidea/delayed_job/blob/v3.0.4/lib/delayed/backend/base.rb#L86 obscures the original exception's stack and makes it really hard to debug. See a conversation in https://discuss.newrelic.com/t/upgrade-to-mongoid-4-x-has-agent-spewing-errors-that-have-no-clear-origin/2372/3. 

I am not sure what the best way to fix this is, but maybe logging the entire original exception + stack is a start?

**Comment by dinkman** [comment]
Yeah, this is highly non-ideal for a developer. Looks like right now everyone has to monkey patch. While not a sexy fix, exposing some default functionality (e.g., turned on via config file) to log inline stack traces would be really useful.
